lib/spack/spack/binary_distribution.py

```diff
@@ -1377,77 +1473,114 @@ def is_backup_file(file):
             relocate.relocate_text(text_names, prefix_to_prefix_text)
 
 
-def extract_tarball(spec, filename, allow_root=False, unsigned=False,
-                    force=False):
-    """
-    extract binary tarball for given package into install area
-    """
-    if os.path.exists(spec.prefix):
-        if force:
-            shutil.rmtree(spec.prefix)
-        else:
-            raise NoOverwriteException(str(spec.prefix))
-
-    tmpdir = tempfile.mkdtemp()
+def _extract_inner_tarball(spec, filename, extract_to, unsigned, remote_checksum):
```

Comment:
This argument is never used, since it's assigned immediately below:
```suggestion
def _extract_inner_tarball(spec, filename, unsigned, remote_checksum):
```